src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -150,6 +150,10 @@ public function decode($data, $format, array $context = array())
         }
         fclose($handle);
 
+        if (isset($context['as_collection']) && $context['as_collection']) {
```

Comment:
Maybe we could use a 3 value switch here (`null`/`true`/`false`) with `null` (or not set) the current behavior (guess), `true` forcing a collection and `false` forcing an unique element.

Reply:
I think it's ok to say `true` should return the collection and false it the old behaviour.

Reply:
`$context['as_collection'] ?? false`?